app/domain/authentication/status_webservice.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'util/fetch_resource'
+
+module Authentication
+  class StatusWebservice
+
+    attr_reader :parent_webservice
+
+    def self.from_webservice(webservice)
+      self.new(webservice)
+    end
+
+    def initialize(parent_webservice)
+      @parent_webservice = parent_webservice
+    end
+
+    def name
+      @name ||= "#{@parent_webservice.name}/status"
+    end
+
+    def resource_id
+      @resource_id ||= "#{@parent_webservice.resource_id}/status"
+    end
+
+    def resource
+      @resource ||= Util::FetchResource.new.(resource_id: resource_id)
+    end
```

Comment:
This DB interaction here is still breaking the value object contract:  Even though we're technically not using `Dry::Struct` anymore, this is still (or should still be) a value object.  That is, the distinction we're trying uphold between value objects and "things with side effects" (typically `CommandClass` instances for us, but not necessarily) is conceptual.

It's fundamentally about how we're partitioning responsibilities of our code, and not about the specific technologies we're using (eg, it's incidental that we're using `Dry::Struct` to implement value objects.  There are many other value object libs, or we could use plain ruby objects).  

I did a search in RubyMine for usages of `annotation`, and I didn't notice *this* `annotation` method being used anywhere.  Perhaps we could just remove it?  If I missed where it was being used, see my earlier comment on `authenticator_status_input` above, and see if an approach like that can be used to remove this.  I think it shouldn't be too hard.

Reply:
created issue https://github.com/cyberark/conjur/issues/1092 for this. it's not a very small change as it changes the way these classes are created and used. we'll get to it in our epic leftovers.